qt_ui/windows/AirWingConfigurationDialog.py

```diff
@@ -79,25 +82,35 @@ class SquadronBaseSelector(QComboBox):
     def __init__(
         self,
         bases: Iterable[ControlPoint],
-        squadron: Squadron,
+        aircraft_type: Optional[AircraftType],
+        current_value: str,
     ) -> None:
         super().__init__()
-        self.bases = list(bases)
-        self.squadron = squadron
         self.setSizeAdjustPolicy(self.AdjustToContents)
+        self.bases = list(bases)
+        self.add_bases_for_aircraft_type(aircraft_type)
+        self.setCurrentText(current_value)
+
+    def add_bases_for_aircraft_type(self, aircraft_type: Optional[AircraftType]):
+        self.clear()
+        if aircraft_type:
+            for base in self.bases:
+                if not base.can_operate(aircraft_type):
+                    continue
+                self.addItem(base.name, base)
+            self.model().sort(0)
+        else:
+            self.addItem("Select aircraft type first", None)
 
-        for base in self.bases:
-            if not base.can_operate(self.squadron.aircraft):
-                continue
-            self.addItem(base.name, base)
-        self.model().sort(0)
-        self.setCurrentText(self.squadron.location.name)
+        self.update()
 
 
 class SquadronConfigurationBox(QGroupBox):
+    remove_squadron_signal = Signal(Squadron)
+
     def __init__(self, squadron: Squadron, theater: ConflictTheater) -> None:
         super().__init__()
-        self.setCheckable(True)
+        self.setCheckable(False)
```

Comment:
This is the default, so can just kill the line.

Reply:
thanks :)